### DISCUSSION TITLE
setBackgroundMessageHandler doesn't work with large JS code.

### DISCUSSION THREAD

**Discussion by AkeleyUA** [invertase/react-native-firebase#5828] [Q&A]
I have a problem getting data-only push notifications. setBackgroundMessageHandler doesn't work on IOS. But I am deleting a piece of my code. setBackgroundMessageHandler starts working. As I understand it, it takes some time to load the JS part of the code. And after this time, the background message handler will receive the message. But if no JS code is loaded, it doesn't work. Are there any ways to control this, increase the time to run the code, or other ways to work around this problem.

**Comment by mikehardy** [comment]
As of react-native-firebase 12.4.0 that should not be happening

https://github.com/invertase/react-native-firebase/pull/5547
https://github.com/invertase/react-native-firebase/pull/5708

https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#1240-2021-07-29

With no reproducible example, it won't be possible to give more advice than to make sure that you are on current versions (or at least, greater than or equal to 12.4.0 here)
